tools/testsys/src/aws_resources.rs

```diff
@@ -227,3 +297,181 @@ impl AwsK8s {
         Ok(Crd::Test(sonobuoy))
     }
 }
+
+/// In order to easily create migration tests for `aws-k8s` variants we need to implement
+/// `Migration` for it.
+impl Migration for AwsK8s {
+    fn migration_config(&self) -> Result<MigrationsConfig> {
+        Ok(MigrationsConfig {
+            tuf_repo: self
+                .tuf_repo
+                .as_ref()
+                .context("Tuf repo metadata is required for upgrade downgrade testing.")?
+                .clone(),
+            starting_version: self
+                .starting_version
+                .as_ref()
+                .context("You must provide a starting version for upgrade downgrade testing.")?
+                .clone(),
+            migrate_to_version: self
+                .migrate_to_version
+                .as_ref()
+                .context("You must provide a target version for upgrade downgrade testing.")?
+                .clone(),
+            region: self.region.to_string(),
+            secrets: self.secrets.clone(),
+            capabilities: self.capabilities.clone(),
+            assume_role: self.assume_role.clone(),
+        })
+    }
+
+    fn instance_provider(&self) -> String {
+        let cluster_name = self.cluster_name();
+        format!("{}-instances", cluster_name)
+    }
```

Comment:
Based on the function name I'd expect this to be providing instances. Do we need this interface or will all of the implementations basically look like this?

Reply:
This will be used for the ecs and VMware workflow as well. It is just the name of the resource object responsible for providing instances.